invenio_rdm_records/records/jsonschemas/records/record-v1.0.0.json

```diff
@@ -1,402 +1,712 @@
+                "description": "A reference string."
+              },
+              "identifier": {"type": "string"},
+              "scheme": {"type": "string"}
+            }
+          }
         }
+      }
+    },
+
+    "metaext": {
+      "type": "object",
+      "description": "Configured additional metadata",
+      "propertyNames": {
+        "pattern": "^[A-Za-z][A-Za-z0-9_]*:[A-Za-z0-9_]+$"
       },
-      "titles": {
-        "description": "Record title.",
-        "type": "array",
-        "minItems": 1,
-        "uniqueItems": true,
-        "items": {
-          "type": "object",
-          "properties": {
-              "title": {
-                "description": "Title of the record.",
-                "type": "string"
-              },
-              "type": {
-                "description": "Type of title.",
-                "type": "string"
-              },
-              "lang": {
-                "description": "Language of the title. ISO 639-3 language code.",
-                "type": "string",
-                "maxLength": 3
-              }
+      "additionalProperties": {
+        "anyOf": [
+          {
+            "type": "array",
+            "items": {
+              "type": ["boolean", "number", "integer", "string"]
+            }
           },
-          "required": ["title"]
+          {"type": "boolean"},
+          {"type": "number"},
+          {"type": "integer"},
+          {"type": "string"}
+        ]
+      }
+    },
+
+    "tombstone": {
+      "type": "object",
+      "description": "Tombstone for the record.",
+      "additionalProperties": false,
+      "properties": {
+        "reason": {
+          "type": "string",
+          "description": "Reason for removal."
+        },
+        "category": {
+          "type": "string",
+          "description": "Category for the removal."
+        },
+        "removed_by": {"$ref": "#/definitions/agent"},
+        "timestamp": {
+          "type": "string",
+          "description": "ISO8601 formatted timestamp in UTC.",
+          "format": "date-time"
         }
-      },
-      "extensions": {
-        "additionalProperties": {
-          "anyOf": [
-            {
-              "type": "array",
-              "items": {
-                "type": ["boolean", "number", "string"]
-              }
+      }
+    },
+
+    "provenance": {
+      "type": "object",
+      "description": "Record provenance.",
+      "additionalProperties": false,
+      "properties": {
+        "created_by": {"$ref": "#/definitions/agent"},
+        "on_behalf_of": {"$ref": "#/definitions/agent"},
+        "contact": {"$ref": "#/definitions/agent"}
+      }
+    },
+
+    "access": {
+      "type": "object",
+      "description": "Record access control and ownership.",
+      "additionalProperties": false,
+      "properties": {
+
+        "metadata": {
+          "description": "Metadata visibility (true - public, false - private)",
+          "type": "boolean"
```

Comment:
Minor: Making these fields boolean flags might be limiting us in the number of available access "modes" we want to implement. I'm aware that the combination of these booleans translates to different access scenarios (i.e. hidden record, closed files only, etc). Of course, adding a new type of access mode could be achieved via just adding another boolean field, but then the number of total states, along with the boolean logic that has to be performed to translate these states becomes more complex.

A use case could be having "partially" hidden metadata fields. There would a couple of ways to implement:

- With boolean flags only: `metadata: true` and `metadata_hidden_fields: false`
- With enums: `metadata: partial`